File: extruturadedados/Revisao/repetição/Adivinhacao.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

int main() {
    srand(time(0));
    int segredo = rand() % 100 + 1;  //
    int chute, tentativas = 0;

    cout << "🎲 Adivinhe o número entre 1 e 100!\n";

    do {
        cout << "Digite seu palpite: ";
        cin >> chute;
        tentativas++;

        if (chute > segredo)
            cout << "Muito alto!\n";
        else if (chute < segredo)
            cout << "Muito baixo!\n";
        else
            cout << "🎉 Acertou em " << tentativas << " tentativas!\n";
    } while (chute != segredo);
}

File: extruturadedados/Revisao/repetição/adivinhacao.cs
using System;

class Program
{
    static void Main()
    {
        Random rand = new Random();
        int segredo = rand.Next(1, 101); // número entre 1 e 100
        int chute;
        int tentativas = 0;

        Console.WriteLine("🎲 Adivinhe o número entre 1 e 100!");

        do
        {
            Console.Write("Digite seu palpite: ");
            string entrada = Console.ReadLine();

            // Tenta converter a entrada para inteiro
            if (!int.TryParse(entrada, out chute))
            {
                Console.WriteLine("Por favor, digite um número válido!");
                continue;
            }

            tentativas++;

            if (chute > segredo)
                Console.WriteLine("Muito alto!");
            else if (chute < segredo)
                Console.WriteLine("Muito baixo!");
            else
                Console.WriteLine($"🎉 Acertou em {tentativas} tentativas!");

        } while (chute != segredo);
    }
}
